File: src/main/java/net/golem/raknet/protocol/connected/ConnectedPongPacket.java
package net.golem.raknet.protocol.connected;

import net.golem.raknet.codec.PacketDecoder;
import net.golem.raknet.codec.PacketEncoder;
import net.golem.raknet.protocol.DataPacket;
import net.golem.raknet.protocol.RakNetPacketIds;

public class ConnectedPongPacket extends DataPacket {

	public long pingTime;

	public long pongTime;

	public ConnectedPongPacket() {
		super(RakNetPacketIds.CONNECTED_PONG);
	}

	@Override
	public void encode(PacketEncoder encoder) {
		encoder.writeLong(pingTime);
		encoder.writeLong(pongTime);
	}

	@Override
	public void decode(PacketDecoder decoder) {
		pingTime = decoder.readLong();
		pongTime = decoder.readLong();
	}

	@Override
	public String toString() {
		return "ConnectedPongPacket{" +
				"pingTime=" + pingTime +
				", pongTime=" + pongTime +
				'}';
	}
}

File: src/main/kotlin/raknet/packet/protocol/connected/ConnectedPongPacket.kt
package raknet.packet.protocol.connected

import io.netty.buffer.ByteBuf
import raknet.packet.DataPacket
import raknet.packet.PacketType

class ConnectedPongPacket(
    var pingTime: Long,
    var pongTime: Long
): DataPacket(PacketType.CONNECTED_PONG.id()) {

    override fun encodeOrder(): Array<Any> {
        return arrayOf(pingTime, pongTime)
    }

    companion object {
        fun from(buffer: ByteBuf): ConnectedPongPacket {
            return ConnectedPongPacket(
                buffer.readLong(),
                buffer.readLong()
            )
        }
    }

    override fun toString(): String {
        return "ConnectedPongPacket(pingTime=$pingTime, pongTime=$pongTime)"
    }
}